frontend/streams/index.js

```diff
@@ -0,0 +1,18 @@
+import { main$ } from '@shopgate/pwa-common/streams/main';
+import {
+  UPDATE_USER,
+  SET_DEFAULT_ADDRESS,
+} from './../constants/ActionTypes';
+
+/**
+ * Gets triggered when we update user data (personal, addresses, etc)
+ * @type {Observable}
+ */
+export const userWillUpdate$ = main$.filter(({ action }) => action.type === UPDATE_USER);
```

Comment:
This might be very confusing to third party developers becuase we have a `userDidUpdate` stream in `pwa-common` which does something differently. Maybe should have instead a userAddresses(Did/Will)Update